src/main/java/seedu/address/storage/JsonAdaptedTask.java

```diff
@@ -0,0 +1,59 @@
+package seedu.address.storage;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonValue;
+
+import seedu.address.commons.exceptions.IllegalValueException;
+import seedu.address.model.animal.Task;
+
+
+
+/**
+ * A Jackson-friendly adapted version of the {@link Task} model class.
+ */
+public class JsonAdaptedTask {
+    private final String description;
+    private final boolean taskStatus;
+
+    /**
+     * Constructs a {@code JsonAdaptedTask} with the given Task details.
+     * This constructor is used internally by Jackson to deserialize {@link Task} JSON objects.
+     */
+    @JsonCreator
+    public JsonAdaptedTask(String description, boolean taskStatus) {
+        this.description = description;
+        this.taskStatus = taskStatus;
+    }
+
+    /**
+     * Converts a given {@code Task} into a {@code JsonAdaptedTask}, which can then be
+     * serialized by Jackson into JSON objects.
+     */
+    public JsonAdaptedTask(Task source) {
+        description = source.getDescription();
+        taskStatus = source.checkIsDone();
+    }
+
+    @JsonValue
+    public String getDescription() {
+        return description;
+    }
+
+    @JsonValue
+    public boolean getIsDone() {
+        return taskStatus;
+    }
```

Comment:
when you implement the storage feature for Task and TaskList, do rename this method to just `isDone()` to reflect a boolean return type.